src/finch/tensor.py

```diff
@@ -38,105 +56,123 @@ def ndim(self) -> int:
         return jl.ndims(self._obj)
 
     @property
-    def shape(self) -> Tuple[int, ...]:
+    def shape(self) -> tuple[int, ...]:
         return jl.size(self._obj)
 
     @property
     def size(self) -> int:
         return np.prod(self.shape)
 
+    @property
+    def _permutation(self) -> tuple[int, ...]:
+        return jl.typeof(self._obj).parameters[1]
+
+    def _apply_new_permutation(self, value: tuple[int, ...]) -> None:
+        p = self._permutation
+        if self.order == self.ROW_MAJOR:
```

Comment:
could we use a utility function that maps e.g. Row_major to [1, 2, 3, ... n] and then handle the general case in this function and others? We could use the function to parse orders any time we accept them.

Reply:
Done! I added `def preprocess_order(...)` that maps string names, NoneType and tuples to 1-indexed permutations.